library/index.c

```diff
@@ -323,24 +323,25 @@ int eblob_index_blocks_fill(struct eblob_base_ctl *bctl)
 }
 
 
-static struct eblob_disk_control *eblob_find_on_disk(struct eblob_backend *b,
-		struct eblob_base_ctl *bctl, struct eblob_disk_control *dc,
+static int eblob_find_on_disk(struct eblob_backend *b,
```

Comment:
No, I am not mistaken.
search_start != start
search_end != end
And these ^ inequalities stays true before and after bsearch() call.

start, end - range of whole index file
search_start, search_index - range of index block

So my previous comment is correct.